### PR TITLE
chore: adjusting dependencies of expressions and routers

### PR DESCRIPTION
**Pull request by 2eha0**
## Simplified Usage and Installation of entities-routes
To streamline the installation process, `optionalDependencies` and `peerDependencies` have been moved to dependencies. Now, the host app only needs to install entities-routes itself.

before:
```json5
// package.json
"@kong-ui-public/entities-routes": "x.x.x"
"@kong-ui-public/forms": "x.x.x"
"@kong-ui-public/expressions": "x.x.x"
```
```js
// vue
import { RouteForm } from '@kong-ui-public/entities-routes'
import '@kong-ui-public/expressions/dist/style.css'
```
after:
```json5
// package.json
"@kong-ui-public/entities-routes": "x.x.x"
```
```js
// vue
import { RouteForm } from '@kong-ui-public/entities-routes'
```
All other dependencies are automatically installed.

## Bundle Size Optimization
By externalizing most of the module from `node_modules`, the bundle size has been reduced from [633KB](https://www.npmjs.com/package/@kong-ui-public/entities-routes/v/3.5.4) to [308KB](https://www.npmjs.com/package/@kong-ui-public/entities-routes/v/3.5.5-pr.1550.a8a375a9.0) (-51.34%)

The `@kong/atc-router` was kept in the expressions bundle because there are no scenarios where it would be used directly in the host app, and its use requires the `wasm` Vite plugin, which would complicate the installation process in the host app.

### CODE REVIEW

File: packages/core/expressions/README.md [hunk 1]
```diff
@@ -59,10 +52,6 @@ Import the component(s) in your host application as well as the package styles:
 
 ```ts
 import { asyncInit, ExpressionsEditor } from '@kong-ui-public/expressions'
-import '@kong-ui-public/expressions/dist/style.css'
```
Comment (nekolab) [inline packages/core/expressions/README.md:62]: Shall we keep this line?

Comment (2eha0) [reply packages/core/expressions/README.md:62]: Yes, added it back

File: packages/entities/entities-routes/vite.config.ts [hunk 2]
```diff
@@ -19,9 +19,13 @@ const config = mergeConfig(sharedViteConfig, defineConfig({
...
-        '@kong-ui-public/expressions', // This is optional if we do not use Expressions features
-        '@kong-ui-public/expressions/dist/style.css', // This is optional if we do not use Expressions features
-        'monaco-editor', // This is optional if we do not use Expressions features
+        '@kong-ui-public/entities-shared',
+        '@kong-ui-public/entities-shared/dist/style.css',
+        '@kong-ui-public/expressions',
+        '@kong-ui-public/expressions/dist/style.css',
+        '@kong/icons',
+        'lodash.isequal',
+        'vue-router',
```
Comment (nekolab) [inline packages/entities/entities-routes/vite.config.ts:28]: `vue-route` is already included in `vite.config.shared.ts`

Comment (2eha0) [reply packages/entities/entities-routes/vite.config.ts]: deleted

**Review by nekolab — APPROVED** [review]
LGTM

**Review by Leopoldthecoder — APPROVED** [review]

**Review by nekolab — APPROVED** [review]
LGTM

**Review by Leopoldthecoder — APPROVED** [review]

File: packages/entities/entities-routes/package.json [hunk 3]
```diff
@@ -71,11 +71,8 @@
   },
   "dependencies": {
     "@kong-ui-public/entities-shared": "workspace:^",
+    "@kong-ui-public/expressions": "workspace:^",
```
Comment (adamdehaven) [inline packages/entities/entities-routes/package.json:74]: This is incorrect; you’re externalizing the dependencies in the Vite config, but then listing them as direct dependencies in the package.json?

**Pull request merged**